python/src/iceberg/expressions/base.py

```diff
@@ -147,3 +148,61 @@ def get(self, container: StructProtocol) -> Any:
             Any: The value at position `self.position` in the container
         """
         return container.get(self.position)
+
+
+class BoundReference:
+    """A reference bound to a field with an accessor for acquiring the field's value
```

Comment:
Accessors are internal, so there is no need to mention them in docs. This doesn't necessarily need to use an accessor to work.

Reply:
I updated this description to "A reference bound to a field in a schema" but I'm not sure I fully understand what you mean by:
> This doesn't necessarily need to use an accessor to work.

Can you describe how this would be used without an accessor? I know this class won't be exposed at the top level API but I'd imagine a developer working in this area of the code would find a description of the relationship to accessors as helpful.

Reply:
I don't think it would work without an accessor right now. But you don't need to tie the implementation to an accessor. That's an internal detail.